modules/um_callerid.c

```diff
@@ -105,7 +117,7 @@ allow_message(struct Client *source_p, struct Client *target_p)
 		return true;
 
 	/* XXX: controversial?  allow opers to send through +g */
-	if (MayHavePrivilege(source_p, "oper:message"))
+	if (IsSetTalkThroughCallerID(source_p))
```

Comment:
This makes it impossible for services to speak to a +g user without hacking the services code.